packages/Search/src/DTK_LinearBVH_decl.hpp

```diff
@@ -95,37 +122,39 @@ BoundingVolumeHierarchy<DeviceType>::BoundingVolumeHierarchy(
     {
         Kokkos::View<size_t *, DeviceType> permutation_indices( "permute", 1 );
         Details::TreeConstruction<DeviceType>::initializeLeafNodes(
-            permutation_indices, primitives, _leaf_nodes );
+            permutation_indices, primitives, getLeafNodes() );
         return;
     }
 
     // determine the bounding box of the scene
     Details::TreeConstruction<DeviceType>::calculateBoundingBoxOfTheScene(
-        // FIXME this->getBoundingVolume( this->getRoot() );
-        primitives, _internal_nodes[0].bounding_box );
+        primitives, getBoundingVolume( getRoot() ) );
 
     // calculate morton code of all objects
     auto const n = primitives.extent( 0 );
     Kokkos::View<unsigned int *, DeviceType> morton_indices(
         Kokkos::ViewAllocateWithoutInitializing( "morton" ), n );
     Details::TreeConstruction<DeviceType>::assignMortonCodes(
-        // FIXME this->getBoundingVolume( this->getRoot() );
-        primitives, morton_indices, _internal_nodes[0].bounding_box );
+        primitives, morton_indices, getBoundingVolume( getRoot() ) );
 
     // sort them along the Z-order space-filling curve
     auto permutation_indices = Details::sortObjects( morton_indices );
 
     Details::TreeConstruction<DeviceType>::initializeLeafNodes(
-        permutation_indices, primitives, _leaf_nodes );
+        permutation_indices, primitives, getLeafNodes() );
+
+    Kokkos::View<int *, DeviceType> parents(
+          Kokkos::ViewAllocateWithoutInitializing( "parents" ),
+          2 * size() - 1 );
```

Comment:
Maybe replace by `_internal_and_leaf_nodes.size()`? I'd like to see as few magic calculations as possible.